src/shared/WithAppBar/index.tsx

```diff
@@ -1,14 +1,43 @@
 import React from "react";
-import { AppBar, Toolbar, Typography } from "@mui/material";
+import { AppBar, Toolbar, Typography, IconButton } from "@mui/material";
 import { Stack } from "./styles";
 
-const withAppBar = (WrappedComponent: React.ComponentType, title: string) => {
+// Обновленная функция withAppBar
+const withAppBar = (
+  WrappedComponent: React.ComponentType,
+  title: string,
+  icon?: string | null
+) => {
   return () => {
+    const handleIconClick = () => {
+      // Перенаправление на страницу настроек
+      window.location.href = "/settings";
```

Comment:
1. What you will do if there will be requirement to redirect not to only settings, but to exercises tab?
2. Don't use window.location. Use react-router methods instead